listenbrainz/webserver/views/stats_api.py

```diff
@@ -151,3 +147,16 @@ def _is_valid_range(stats_range):
         result (bool): True if given range is valid
     """
     return stats_range in StatisticsRange.__members__
+
+
+def _get_stats_for_range(data: UserArtistStatRow, stats_range: StatisticsRange) -> UserArtistStatForRange:
+    if stats_range == 'all_time':
```

Comment:
@alastair any suggestions here? this function seems non ideal, it was just `data['artist'][stats_range]` before this, also, i'm wary of doing this for release and recording as well...

Reply:
I may be wrong, but can't we convert `data` to a dictionary and revert back to the original code.
Something like, `return dict(data)['artist'][stats_range]`

Reply:
in this case you're passing a `str` into this method instead of a `StatisticsRange`. What if we did validation in the view to ensure that the parameter was a valid value for the enum? Then we could maybe do something like

`getattr(data.artist, stats_range.value)`

because all of the validation is already done. You could also do `getattr(data.artist, stats_range.value, None)` and raise the ApiBadRequest if the value is None

Reply:
Used getattr in some places, in some places I thought it was better to just be more explicit and duplicate code.